### DISCUSSION TITLE
Expose deBounce of search field in admin UI + allow to disable search as you type ?

### DISCUSSION THREAD

**Discussion by geminigeek** [payloadcms/payload#1030] [Feature Requests & Ideas]
hi,

i am testing payload to build a very large db/application, i just inserted 11 Million records. i would like to suggest these features / options

1. i need to disable  "search as you type" in admin backend , only want to click button and search.  
2.  need to have  deBounce  exposed per field or per collection to modify search behavior ?

thanks

**Comment by jmikrut** [comment]
Hey @geminigeek — this is certainly doable! I am considering ways that this could be semantic enough and / or usable enough to merit a config property and I'm thinking that you might actually be better off by just using a custom `List` component for your big collection if you want to do this, as the built-in filtering within the List view also searches _**as you type**_. So disabling that only on the top level search would only get you so far.

Could you swap in a custom list component instead? Pull the items that you need from the built-in Payload list component, and customize accordingly?

**Reply by geminigeek** [reply]
Hi James,

i am very new to Js and frontend , i am coming from joomla/php background , i am a bit good with Vue and Astro , not with react, currently i cant make Ui component on my own :(

as far as your comment, i think the default ui payload provide is excellent, i would suggest a further suggestion to expose custom endpoint for search functionality and let user decide how to search or use a third party search system, just send a post request to a custom endpoint with all filters options selected by user , as the ui expects a certain fields in search result, the custom endpoint will send the same. this way we can implement 3rd party search in backend and can use frontend search functionality in backend as well !!

thanks for reading

**Reply by geminigeek** [reply]
hi,

one more quick performance suggestion. on pagination in find query this can make find 10X faster 

1. Option to have just next and previous available in find results , with pagination : true we get count , but some time we don't want count , we just need to know if we have a next page !!
2. this can be done by querying for n+1 items , n being the per page limit or default page Limit , after we get result  pop the last search result item before sending search results. and set the flag hasNextPage:true
3. if current page is more than 1 hasPrevPage can be defaulted to true.

**Reply by DanRibbens** [reply]
I'm thrilled that you are using Payload with collections this large! We absolutely want to make Payload capable of handling such loads.

I'm verify familiar with how other frameworks/ORMs deal with this. I also have a PHP background and in Laravel you have `simplePaginate()` for the purpose you described. I like this idea for Payload.

I will do some digging to plan out how to best proceed on these enhancements.
